### PR TITLE
QoL improvements for birthday announcer

### PR DESCRIPTION
**Pull request by zwliew**
## Changes
- Update Discord.js to v12.5.1
- Fix ESLint glob parameter to support zsh on macOS
- Announce usernames instead of tagging user IDs (fixes the @invalid-user bug)
- Use some preset reactions
- Prettify the date printed -- from DD/MM to DD MMM (e.g. 03/01 -> 03 Jan)
- Support dates inputted in D{1,2}/M{1,2} format
- Reject date inputs that are out of range

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -10,7 +10,7 @@
     "dev": "docker-compose -f docker-compose.yml up --build",
     "docker-dev": "ts-node-dev --transpile-only --poll src/main/App.ts",
     "build": "rimraf -rf build/ && tsc -p .",
-    "lint": "eslint ./src/**"
+    "lint": "eslint \"./src/**\""
```
Comment (zwliew) [inline package.json:13]: I needed to do this for the command to work on zsh on Mac

File: src/main/App.ts [hunk 2]
```diff
@@ -80,17 +80,20 @@ export class App {
         });
 
         // eslint-disable-next-line @typescript-eslint/no-unused-vars
-        this.bot.on(App.MESSAGE_UPDATE, (oldMessage: Message, newMessage: Message): void => {
+        this.bot.on(App.MESSAGE_UPDATE, async (_: Message | PartialMessage, newMessage: Message | PartialMessage): Promise<void> => {
+            if (newMessage.partial) {
```
Comment (zwliew) [inline src/main/App.ts:84]: seems hacky...

**Comment by zwliew** [comment]
@andrewome senpai, review please and thank you!

File: src/main/command/birthdaycommands/DeleteBirthdayCommand.ts [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    /**
+     * Displays all the users with a birthday on the date given as an argument.
+     *
+     * @param { CommandArgs } commandArgs
+     * @returns CommandResult
+     */
+    public async execute(commandArgs: CommandArgs): Promise<CommandResult> {
+        const { userId, messageReply, server } = commandArgs;
+
+        if (!userId) {
```
Comment (andrewome) [inline src/main/command/birthdaycommands/DeleteBirthdayCommand.ts:24]: `userId` will be truthy. It's the author of the message sent. Just assert it using the `!` operator, then we don't need the invalid embed part already.

Comment (andrewome) [reply src/main/command/birthdaycommands/DeleteBirthdayCommand.ts:24]: I made it optional in `CommandArgs` so I could use dependency injection for testing 😬

File: src/main/command/birthdaycommands/DeleteBirthdayCommand.ts [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+            await messageReply(this.generateInvalidEmbed());
+            return UNSUCCESSFUL_COMMANDRESULT;
+        }
+
+        deleteBirthday(server.serverId, userId);
```
Comment (andrewome) [inline src/main/command/birthdaycommands/DeleteBirthdayCommand.ts:29]: You may consider adding a failing message if nothing was deleted from the db. You can detect this from the return object of the `statement#run` statement.
But I think it's not necessary lol

https://github.com/JoshuaWise/better-sqlite3/blob/master/docs/api.md#runbindparameters---object
^ `statement#run` returns an object with a `changes` key which shows the total number of rows that were inserted, updated, or deleted by this operation.

**Review by andrewome — CHANGES_REQUESTED** [review]

File: src/main/command/birthdaycommands/DeleteBirthdayCommand.ts (hunk 4 above)
Comment (zwliew) [reply src/main/command/birthdaycommands/DeleteBirthdayCommand.ts:29]: Good point. I'll add it soon.

**Pull request merged**